apps/ensindexer/src/handlers/Registry.ts

```diff
@@ -124,7 +139,19 @@ export const makeRegistryHandlers = (ownedName: OwnedName) => {
 
           // attempt to heal the label associated with labelhash via ENSRainbow
           // https://github.com/ensdomains/ens-subgraph/blob/c68a889/src/ensRegistry.ts#L112-L116
-          const healedLabel = await labelByHash(labelhash);
+          let healedLabel = await labelByHash(labelhash);
+
+          // if the label has not healed with ENSRainbow query
+          // and healing label from reverse addresses is enabled, give it a go
+          if (!healedLabel && canHealReverseAddresses()) {
```

Comment:
maybe check that first and if not try healing with ENSRainbow

Reply:
Ah ok, so first we try healing based of the in memory data (using reverse address method), and only if it didn't return any result we should try healing with ENSRainbow. Did I get that right?

Reply:
Yes, maybe it will be faster?